File: Workshop/JavaBasicsProject/src/Basics/FileIO/CountLines/CountLines.java
package Basics.FileIO.CountLines;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class CountLines {
  public static void main(String[] args) {
    // Write a function that takes a filename as string as a parameter
    // and returns the number of lines the file contains.
    // It should return zero if it can't open the file
    // and should not raise any error.


    String filename = "text.txt";
    int numberOfLines = getNumberOfLines(filename);

    //test
    System.out.println(numberOfLines);

  }

  private static int getNumberOfLines(String fileName) {
    Path filePath = Paths.get("src/Basics.FileIO/CountLines/" + fileName);
    try {
      return Files.readAllLines(filePath).size();
    } catch (IOException e) {
      return 0;
    }
  }
}

File: Foundation/week_02-week_04/PythonBasicsProject/week_03/file_io/count_lines/count_lines.py
# Write a function that takes a filename as string as a parameter
# and  returns the number of lines the file contains.
# It should return zero if it can't open the file and
# should not raise any error.

def number_of_lines(file_name):
    try:
        my_file = open(file_name, 'r')
        lines = my_file.readlines()
        my_file.close()
        return len(lines)
    except IOError:
        return 0


print(number_of_lines("asd.txt"))
